RequestService/RequestService.Handlers/PostRequestForHelpHandler.cs

```diff
@@ -0,0 +1,369 @@
+            bool multiVolunteers,
+            bool repeat, CancellationToken cancellationToken)
+        {
+            var actions = _groupService.GetNewRequestActions(new GetNewRequestActionsRequest()
+            {
+                HelpRequest = helpRequestDetail.HelpRequest,
+                NewJobsRequest = helpRequestDetail.NewJobsRequest
+            }, cancellationToken).Result;
+
+            if (actions == null)
+            {
+                throw new Exception("No new request actions returned");
+            }
+
+            int requestId = await _repository.NewHelpRequestAsync(new PostNewRequestForHelpRequest()
+            {
+                HelpRequest = helpRequestDetail.HelpRequest,
+                NewJobsRequest = helpRequestDetail.NewJobsRequest
+            }, fulfillable, formVariant.RequestorDefinedByGroup, suppressRecipientPersonalDetails, multiVolunteers, repeat);
+
+            if (requestId == 0)
+            {
+                throw new Exception("Error in saving request");
+            }
+
+            await ProcessRequestActions(actions, requestId, cancellationToken);
+
+            return requestId;
+        }
+
+        public async Task<PostRequestForHelpResponse> Handle(PostRequestForHelpRequest request, CancellationToken cancellationToken)
+        {
+            PostRequestForHelpResponse response = new PostRequestForHelpResponse()
+            {
+                RequestIDs = new List<int>()
+            };
+            var firstHelpRequestDetail = request.HelpRequestDetails.First();
+            bool invalidPostCode = await InvalidPostCode(firstHelpRequestDetail.HelpRequest, cancellationToken);
+
+            if(invalidPostCode)
+            {
+                return GetPostRequestForHelpResponse(new List<int> { -1 }, Fulfillable.Rejected_InvalidPostcode);
+            }
+
+            bool multiVolunteers = firstHelpRequestDetail.MultiVolunteer;
+            bool repeat = firstHelpRequestDetail.Repeat;
+
+            HandleASAP(request);
+            AddMultiAndRepeats(request);
+           
+
+            var json = JsonConvert.SerializeObject(request);
+
+            var requestId = await _repository.GetRequestIDFromGuid(firstHelpRequestDetail.HelpRequest.Guid);
+
+            if (requestId > 0)
+            {
+                return GetPostRequestForHelpResponse(new List<int> { requestId }, Fulfillable.Accepted_ManualReferral);
+            }
+
+            var formVariant = await _groupService.GetRequestHelpFormVariant(firstHelpRequestDetail.HelpRequest.ReferringGroupId, firstHelpRequestDetail.HelpRequest.Source, cancellationToken);
+
+            if (formVariant == null)
+            {
+                return GetPostRequestForHelpResponse(new List<int> { -1 }, Fulfillable.Rejected_ConfigurationError);
+            }
+
+            var failedChecks = await FailedChecks(formVariant.AccessRestrictedByRole, firstHelpRequestDetail.HelpRequest, cancellationToken);
+
+            if(failedChecks)
```

Comment:
Small point, but I'd prefer to see this check nearer the top of the method.

The check for an existing request with the same guid could also be before AddMultiAndRepeats etc.

(I think it's good practise to avoid unnecessary work ;-) )